pkg/apiserver/webhooks/datavolume-validate_test.go

```diff
@@ -455,6 +459,34 @@ func newMultistageDataVolume(name string, final bool, checkpoints []string) *cdi
 		previous = checkpoint
 	}
 
+	var source *cdiv1.DataVolumeSource
+	if sourceType == "vddk" {
```

Comment:
Just a bit of food for thought, would it make sense to have a 'sourceFunc' that returns the source data used in the test. Something like:

```go
vddkSource := func() *cdiv1.DataVolumeSource {
  return &cdiv1.DataVolumeSource{
    VDDK: &cdiv1.DataVolumeSourceVDDK{
      BackingFile: "disk.img",
      URL:         "http://example.com/data",
      UUID:        "12345",
      Thumbprint:  "aa:bb:cc",
      SecretRef:   "secret",
    }
}

imageIOSource := func() *cdiv1.DataVolumeSource {
  return &cdiv1.DataVolumeSource{
    Imageio: &cdiv1.DataVolumeSourceImageIO{
      URL:           "http://example.com/data",
      DiskID:        "disk-123",
      SecretRef:     "secret",
      CertConfigMap: "certs",
    },
}

blankSource := func() *cdiv1.DataVolumeSource {
  return &cdiv1.DataVolumeSource{
    Blank: &cdiv1.DataVolumeBlankImage{},
  }
}

func newMultistageDataVolume(name string, final bool, checkpoints []string, sourceFunc func() *cdiv1.DataVolumeSource) * cdiv1.DataVolume {
...
   source := sourceFunc()
...
}
```
And then we just pass the different functions into the table based on what we are testing. That eliminates the if else, else if, else if, else from here

Reply:
Sure, done.